src/Functions/seriesPeriodDetect.cpp

```diff
@@ -0,0 +1,152 @@
+#pragma clang diagnostic ignored "-Wextra-semi-stmt"
+#pragma clang diagnostic ignored "-Wzero-as-null-pointer-constant"
+#include <pocketfft_hdronly.h>
+#pragma clang diagnostic pop
+
+#include <Functions/IFunction.h>
+#include <Functions/FunctionFactory.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Columns/ColumnsNumber.h>
+#include <Columns/ColumnArray.h>
+
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+    extern const int ILLEGAL_COLUMN;
+}
+
+class FunctionSeriesPeriodDetect : public IFunction
+{
+public:
+    static constexpr auto name = "seriesPeriodDetect";
+
+    static FunctionPtr create(ContextPtr)
+    {
+        return std::make_shared<FunctionSeriesPeriodDetect>();
+    }
+
+    std::string getName() const override
+    {
+        return name;
+    }
+
+    size_t getNumberOfArguments() const override { return 1; }
+
+    bool isVariadic() const override { return false; }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & /*arguments*/) const override
+    {
+        return std::make_shared<DataTypeFloat64>();
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t) const override
+    {
+        ColumnPtr array_ptr = arguments[0].column;
+        const ColumnArray * array = checkAndGetColumn<ColumnArray>(array_ptr.get());
+        const IColumn & src_data = array->getData();
+        //const ColumnArray::Offsets & offsets = array->getOffsets();
+
+        /* const ColumnPtr & col = arguments[0].column;
+        const ColumnVector<Int64> * src_col = checkAndGetColumn<ColumnVector<Int64>>(col.get());
+        const PaddedPODArray<Int64> & values = src_col->getData(); */
+
+        
+
+        auto res = ColumnFloat64::create(1);
+        auto & res_data = res->getData();
+
+        Float64 period;
+
+        if(executeNumber<UInt8>(src_data,  period)
+        || executeNumber<UInt16>(src_data, period)
+        || executeNumber<UInt32>(src_data,  period)
+        || executeNumber<UInt64>(src_data,  period)
+        || executeNumber<Int8>(src_data,  period)
+        || executeNumber<Int16>(src_data,  period)
+        || executeNumber<Int32>(src_data,  period)
+        || executeNumber<Int64>(src_data,  period)
+        || executeNumber<Float32>(src_data,  period)
+        || executeNumber<Float64>(src_data,  period)){
+            res_data[0] = period;
+            return res;
+        }
+        else
+            throw Exception(
+                ErrorCodes::ILLEGAL_COLUMN,
+                "Illegal column {} of first argument of function {}",
+                    arguments[0].column->getName(), getName());
+    }
+
+    template <typename T>  
+    bool executeNumber(const IColumn & src_data, Float64 &period) const
```

Comment:
Please document this function and the logic below for those not familiar with Fourier transform.
A reference to the algorithm would be great.